### ISSUE TITLE
[Docs][A11y] SideNav should behave as a modal overlay when expanded in responsive mode

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#1057]
## 🙋 Documentation Request

SideNav in documentation should behave as a modal overlay when expanded in responsive mode.

In responsive mode, when the SideNav can be toggled using the "hamburger" button:

1. SideNav should have `visibility: hidden` when it is collapsed, so that items within the collapsed SideNav are not included in the tab navigation order for the page.
2. Pressing hamburger button to expand the SideNav should:
  a. Expand the SideNav by adding `docsStyle.visible` className.
  b. Set `aria-hidden="true"` to `main` landmark region and `ThemeSwitcher` element.
  c. Set `tabIndex = -1` on `button` element within `ThemeSwitcher`.
  d. Set focus to the first tabbable descendant of the SideNav overlay.
3. The SideNav should trap focus so that pressing Tab on the last link will focus the first button or link in the SideNav, and pressing Shift+Tab with focus on the first button or link will focus the last button or link in the SideNav.
4. Pressing Esc key should collapse the SideNav and restore focus to the hamburger button.
5. Collapsing the SideNav, or resizing the browser window so that the responsive media query no longer matches, should:
  a. Collapse the SideNav by removing `docsStyle.visible` className.
  b. Remove `aria-hidden` from `main` landmark region and `ThemeSwitcher` element.
  c. Remove `tabIndex` from `button` element within `ThemeSwitcher`.

## 🧢 Your Company/Team

Adobe Accessibility
